src/valueSelectionManager.ts

```diff
@@ -48,10 +48,22 @@ module powerbi.extensibility.visual {
         private getSelectionIds: (value: T | T[]) => ISelectionId[];
         private selectionManager: ISelectionManager;
 
-        public constructor(visualHost: IVisualHost, getSelectionIds: (value: T) => ISelectionId[]) {
+        public constructor(visualHost: IVisualHost, getSelectionIds: (value: T) => ISelectionId[], getDataPoints: () => WordCloudDataPoint[], renderSelection: () => void) {
             this.visualHost = visualHost;
             this.selectionManager = visualHost.createSelectionManager();
 
+            this.selectionManager.registerOnSelectCallback((ids: ISelectionId[]) => {
+                this.clear(false);
+                ids.forEach( (selection: ISelectionId ) => {
+                    getDataPoints().forEach( (dataPoint: WordCloudDataPoint) => {
+                        if (selection.getKey() === dataPoint.selectionIds[0].getKey()) {
```

Comment:
It'd be better to use `selection.includes(dataPoint.selectionIds[0])` method to compare selectors